CMakeLists.txt

```diff
@@ -409,6 +409,9 @@ add_library_variants(aarch64-none-elf picolibc
     aarch64         ""                  "-march=armv8-a"
 )
 add_library_variants(arm-none-eabi picolibc
+    armv4t          ""                  "-march=armv4t"
+    armv5te         ""                  "-march=armv5te"
+    armv6           ""                  "-march=armv6"
```

Comment:
What's the motivation for adding each of these?
There's a cost to building and testing each variant (I personally spend a significant amount of time waiting while this happens) so I want to be sure it benefits someone.

Reply:
If `v4t` covers all those use cases then could you reduce it to adding just that one? It sounds like v5te and v6 would be optimisations.

Reply:
Between 5te and 6 I agree that the difference would be minimal. But between v4t and v5te there have been enough improvements in how arm-thumb interworking is done (every such function call would be affected) that I think it makes sense to have the two variants. Also because people in embedded are quite sensitive about code size.

I'm a little bit hesitant about the optics of having people building a v6 app with a v5 library. They should be clued up enough about the compatibility story that they feel secure enough to use the v5 variant. Which I don't think is a given. I guess we can give a note of it in the documentation, and as we don't yet have a real-world use case, it's probably fair enough to omit it.

Reply:
OK, v4t and v5te it is. v6 can always be added later if a need is identified.
Once [multilib](https://discourse.llvm.org/t/rfc-multilib/67494) eventually lands people won't have to think about which library they should use :)